optional/Optional.py

```diff
@@ -0,0 +1,106 @@
+from abc import abstractmethod
+from .compatible_abc import CompatibleABC
+
+
+def of(thing):
```

Comment:
this feels a bit odd to me.  I get that using the capital "O" Optional for filename it reads like you are doing:
```
Optional.of("Foo")
```
but that file naming is unconventional.  I still wonder about putting the static constructors into the Optional base class?

Reply:
I don't want them inherited by the derived classes. I tried a little __init__.py magic

Reply:
I ended up making a class called Optional with just the two static methods. No weird filenames, nothing weird in __init__.py